static/shared/js/fenced_code.js

```diff
@@ -30,14 +33,21 @@ let stash_func = function (text) {
     return text;
 };
 
-export function wrap_code(code) {
+export function wrap_code(code, lang) {
+    let header = '<div class="codehilite"><pre><span></span><code>';
+    // Mimics the backend logic of adding a data-attribute (data-code-language)
+    // to know what Pygments language was used to highlight this code block.
+    if (lang !== undefined && lang !== "") {
+        const pygments_language_info = pygments_data.langs[lang];
+        let code_language = lang;
```

Comment:
Do we need to HTML-escape `code_language`?  Might be best to do it or add a comment on what it's not necessary.